packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -481,21 +497,32 @@ function dc(
     if (Ctor == null) {
         return null;
     }
+    let props = data.props;
+    let ctor: LightningElementConstructor;
     if (!isComponentConstructor(Ctor)) {
-        throw new Error(`Invalid LWC Constructor ${toString(Ctor)} for custom element <${sel}>.`);
+        if (isLightningElementConfig(Ctor, dereference)) {
+            props = { ...data.props, ...dereference(Ctor, 'props') };
+            ctor = dereference(Ctor, 'ctor') as LightningElementConstructor;
+        } else {
+            throw new Error(
+                `Invalid LWC Constructor ${toString(Ctor)} for custom element <${sel}>.`
+            );
```

Comment:
Can we add a test to verify that this error is thrown? It looks like we don't have one, even today.